src/Authentication/Actions/ActionInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php namespace Sparks\Shield\Authentication\Actions;
+
+use CodeIgniter\HTTP\IncomingRequest;
+
+/**
+ * Interface ActionInterface
+ *
+ * Authentication Actions are steps that can happen after
+ * the main authentication steps, like registration and login.
+ * They can be email activation steps, SMS-based 2FA, etc.
+ *
+ * @package Sparks\Shield\Authentication\Actions
+ */
+interface ActionInterface
```

Comment:
I know this is still WIP so maybe you're figuring this out still, but the looseness of this interface makes me a little uneasy. It looks like we could safely return a `ResponseInterface` from these methods, which makes me wonder if this should be an extension of `RequestHandlerInterface` and part of a legit PSR-15 middleware (https://www.php-fig.org/psr/psr-15/). 

That might be putting the cart before the horse though, but with the framework moving toward more PSR compliance in the HTTP layer this could be a good chance to model interactions with it.

Reply:
PSR-15 is close, but only designed to handle a single response and would make this more complex I think, since my current thinking is a single Action class provides 3 responses: 

1. Display the initial notification/form
2. Handle the form response/send the email/whatever, and display another view to tell the user it's been sent
3. Handle the final action that is probably triggered by entering the code from step 2, or clicking the link in the email, or whatever. 

And middleware definitely won't work in CI since we have to support both route-based and legacy auto-routing, which is why Controller Filters exist instead. :)

Reply:
Likely I will end up requiring a Response as the return value, I just want to make sure it all actually works the way I think it will :)